### PR TITLE
feature: Add commitizen configuration

### PR DESCRIPTION
**Pull request by FelixAbrahamsson**
Adds commitizen to template. Right now only single-line commit messages are allowed, do we want this?

### CODE REVIEW

File: template/{{cookiecutter.repository_name}}/requirements.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ cachetools==4.1.0
 certifi==2020.4.5.1
 chardet==3.0.4
 click==7.1.2
+commitizen==2.1.0
```
Comment (Jim-Holmstroem) [inline template/{{cookiecutter.repository_name}}/requirements.txt:11]: should this be a requirement?

Comment (FelixAbrahamsson) [reply template/{{cookiecutter.repository_name}}/requirements.txt:11]: It's a requirement in the template, not for ml-workflow.

**Review by samedii — COMMENTED** [review]
Very nice :D it should be used on datastream och workflow imo

Commit message is capitalized now unlike the example in the config

Bump pattern will not work with publish.sh?

File: template/{{cookiecutter.repository_name}}/requirements.txt (hunk 1 above)
Comment (Jim-Holmstroem) [reply template/{{cookiecutter.repository_name}}/requirements.txt:11]: yes, still the same question
there is a difference between that we enforce it on our stuff or that we enforce it on anyone creating a template

Comment (samedii) [reply template/{{cookiecutter.repository_name}}/requirements.txt:11]: The template is first and foremost for us to be effective in developing and setting our code standard imo

**Comment by samedii** [comment]
Should allow more than single line probably?

**Comment by Jim-Holmstroem** [comment]
I personally use multiline fairly often, and should probably use them even more often :P

**Comment by FelixAbrahamsson** [comment]
Will look into possibility of writing multiline messages. Thoughts on capitalized vs lower case messages?

**Comment by FelixAbrahamsson** [comment]
Not 100% what's necessary to make it work with publish.sh. I think commitizen automatically tags commits, so we want publish to look for tags instead of generating them? I'm chaning the bum patterns to 
```
bump_pattern = "^(break|feature|fix|tune)"
bump_map = {"break" = "MAJOR", "feature" = "MINOR", "fix" = "PATCH", "tune" = "PATCH"}
```
Anything else that should be in there?

**Comment by samedii** [comment]
> Will look into possibility of writing multiline messages. Thoughts on capitalized vs lower case messages?

We should consider what happens after a full stop, should the next letter be capitalized? Also multiline?

`feature: add commitzen configuration. remove old linting`
`feature: Add commitzen configuration. Remove old linting`
`Feature: Add commitzen configuration. Remove old linting`

Conventional commits are saying lowercase by default and I think I prefer it too

Multiline case:
```
feature!: add commitzen configuration. remove old linting

only allows single line currently. focus on linting is reduced and
we will add more complex configuration in the future.

BREAKING CHANGE
```

Also looks better to me than:
```
feature!: Add commitzen configuration. Remove old linting

Only allows single line currently. Focus on linting is reduced and
we will add more complex configuration in the future.

BREAKING CHANGE
```

**Comment by samedii** [comment]
> ```
> bump_pattern = "^(break|feature|fix|tune)"
> ...
> Anything else that should be in there?

Will it looks for these patterns anywhere or just in the tag?

**Comment by FelixAbrahamsson** [comment]
I like lower case messages as well. However, if we want to allow `feature!` tags we need to explicitly put them in the config which makes the options list very bloated. I suggest the format `feature: !message` instead.

I was gonna message change the format to
```
tag: headline message
<optional> possible multiline body
```
So you are prompted for a mandatory commit tag and headline message, and also an optional message body.

**Comment by FelixAbrahamsson** [comment]
> Will it looks for these patterns anywhere or just in the tag?

According to docs it checks the whole message.

**Comment by samedii** [comment]
Discussed on slack and commitizen seems to not allow the configuration we wanted from conventional commits. Let's close this for now

**Pull request closed without merging**